iron-overlay-manager.html

```diff
@@ -78,10 +78,58 @@
     },
 
     /**
+     * Brings the overlay at the specified index to the front.
+     * @param {number|Element} i
+     */
+    bringToFront: function(i) {
+      // If not a number, assume i is the overlay.
+      if (typeof i !== 'number') {
+        i = this._overlays.indexOf(i);
+      }
+      var overlay = this._overlays[i];
+      var lastI = this._overlays.length - 1;
+      // If already the top element, return.
+      if (!overlay || i === lastI) {
+        return;
+      }
+      // Update z-index to be on top.
+      var minimumZ = Math.max(this.currentOverlayZ(), this._minimumZ);
+      if (this._getZ(overlay) <= minimumZ) {
+        this._applyOverlayZ(overlay, minimumZ);
+      }
+
+      // Shift other overlays behind the new on top.
+      while (i < lastI) {
+        this._overlays[i] = this._overlays[i + 1];
```

Comment:
Uhm, our `move2` is 4x faster than `move1` in general conditions (e.g. move an element that has low index), but incredibly faster for elements with high index http://jsperf.com/bring-to-front
This is gonna be a differentiator for features like `always-on-top`, because the top overlay will most likely have a high index in the array.